pkg/handlers/app.go

```diff
@@ -230,11 +230,24 @@ func responseAppFromApp(a *apptypes.App) (*types.ResponseApp, error) {
 	}
 	d := downstreams[0]
 
-	appVersions, err := store.GetStore().GetDownstreamVersions(a.ID, d.ClusterID, true)
+	appVersions, err := store.GetStore().GetDownstreamVersions(a.ID, d.ClusterID, false)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get downstream versions")
 	}
-	latestVersion := appVersions.AllVersions[0]
+
+	// "latest" is the version that can be viewed and configured (and possibly has been deployed)
+	var latestVersion *downstreamtypes.DownstreamVersion
+	for _, version := range appVersions.AllVersions {
+		if version.Status != storetypes.VersionPendingDownload {
```

Comment:
That's the correct reading.  It replaces the line above `latestVersion := appVersions.AllVersions[0]`.  This is needed because of the true -> false change in `GetDownstreamVersions` call.  And that's needed because when updates are available but not returned from this handler, the UI doesn't behave right.